### PR TITLE
Adds button to re-hide previous events and new event day algorithm

### LINKED ISSUE #16: Should preDays be based on end_time of events?
As pondered by @myth in Slack. This would fix the #15 by removing the problem entirely. This allows for seeing ongoing events as future events while still hiding events that have passed.

### LINKED ISSUE #20: Add a button to re-hide previous events
I'm sure this would rarely be used, but it doesn't seem like all that much work and somebody _could_ want to use it.

### PR DESCRIPTION
**Pull request by plusk**
Adds a button to re-hide previous events that sits between preDays and postDays, if there are any preDays to speak of. It's the same button used for hiding, but now it toggles instead. This resolves #20

New to this PR is a change to how events are organized into days. Previously all events of a day would remain as a "future event" despite perhaps both the start and end date of the events having passed. With this PR, the partitioning is a bit more clever. We now instead look at end dates and split accordingly. If you're in the middle of a day with two events, and the first event has ended, then that event will now be in "past events". In other words, this closes #16 

Also I fixed a bug with the active event feature that I probably caused earlier without anyone noticing. :speak_no_evil: 

For what it's worth, I _tried_ to split the code up more by moving partitionEvents out of the main logic block, but splitting things up much more than this would mean passing parameters to functions that seemed somewhat unreasonable.

### CODE REVIEW

File: components/calendar/index.js [hunk 1]
```diff
@@ -53,55 +55,73 @@ const Calendar = React.createClass({
...
+    if (pastEvents.length) {
+      preDay = <Day events={pastEvents} active={active} eventClickHandler={this.eventClickHandler}/>;
+    }
+    if (futureEvents.length) {
+      postDay = <Day events={futureEvents} active={active} eventClickHandler={this.eventClickHandler}/>;
+    }
+
+    return {
+      preDay: preDay,
+      postDay: postDay
```
Comment (duvholt) [inline components/calendar/index.js]: This can be shortened to { preDay, postDay }. 

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer

File: components/calendar/index.js [hunk 2]
```diff
@@ -122,6 +147,7 @@ const Calendar = React.createClass({
 
     return {
       postDaysSection: postDays,
+      toggleCalendarSection: toggleCalendarSection,
       preDaysSection: preDaysSection
```
Comment (duvholt) [inline components/calendar/index.js]: Shorten this too.

File: components/calendar/index.js [hunk 3]
```diff
@@ -110,9 +130,14 @@ const Calendar = React.createClass({
           { preDays }
         </div>
       );
+      toggleCalendarSection = (
+        <button className="cal-button--preDays" onClick={this.preDaysClickHandler}>
+          Skjul tidligere arrangementer
+        </button>
```
Comment (duvholt) [inline components/calendar/index.js]: Most of `toggleCalendarSection` seems to be repeated. Maybe just find a way to toggle the text instead?

**Review by duvholt — APPROVED** [review]
Good improvement!

File: components/calendar/index.js (hunk 3 above)
Comment (plusk) [reply components/calendar/index.js]: You're absolutely right, I didn't think of that!

File: components/calendar/index.js (hunk 1 above)
Comment (plusk) [reply components/calendar/index.js]: The more you know!
